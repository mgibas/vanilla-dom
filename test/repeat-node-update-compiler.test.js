const compiler = require('../src/repeat-node-update-compiler')
let tagNodeCompiler = require('../src/tag-node-compiler')
let childrenCompiler = jest.fn()
tagNodeCompiler.compile = jest.fn()
childrenCompiler.mockImplementation((a,b) => `CHILDREN_COMPILER_PART('${a}', '${b}')`)
tagNodeCompiler.compile.mockImplementation((a,b,c) => `TAG_NODE_COMPILER_PART('${a}', '${b}', '${c}')`)

describe('repeat-node-update-compiler', () => {
  let options = {
    as:'items',
    index:'i',
    state:'st'
  }

  describe('on compile', () => {
    it('single path', () => {
      let parsed = {
        paths: ['bars'],
        value: () => 'st.bars'
      }
      expect(compiler.compile('parent', {}, parsed, options))
        .toMatchSnapshot();
    })
    it('multipl paths', () => {
      let parsed = {
        paths: ['bars', 'foos'],
        value: () => 'st.bars.push(...st.foos)'
      }
      expect(compiler.compile('parent', {}, parsed, options))
        .toMatchSnapshot();
    })
  })
})
