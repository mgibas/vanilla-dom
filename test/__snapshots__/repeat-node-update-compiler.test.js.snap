// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repeat-node-update-compiler on compile multipl paths 1`] = `
"_updaters.push({
  update: st => {
    let prevCount = items.length;
    items = st.bars.push(...st.foos);
    for (let __removeIndex = prevCount - 1; __removeIndex >= items.length; __removeIndex--) {
      parent.removeChild(_nodes[__removeIndex].node);
    }
    if (prevCount > items.length)
      _nodes.length = items.length;
    for (let __index = 0; __index < _nodes.length; __index++) {
      for (let __updatersIndex = 0; __updatersIndex < _nodes[__index].updaters.length; __updatersIndex++) {
        _nodes[__index].updaters[__updatersIndex].update(st);
      }
    }
    _nodes.length = items.length;
    for (let i = prevCount; i < items.length; i++) {
      _createChildren(i);
    }
  }
});"
`;

exports[`repeat-node-update-compiler on compile single path 1`] = `
"_updaters.push({
  update: st => {
    let prevCount = items.length;
    items = st.bars;
    for (let __removeIndex = prevCount - 1; __removeIndex >= items.length; __removeIndex--) {
      parent.removeChild(_nodes[__removeIndex].node);
    }
    if (prevCount > items.length)
      _nodes.length = items.length;
    for (let __index = 0; __index < _nodes.length; __index++) {
      for (let __updatersIndex = 0; __updatersIndex < _nodes[__index].updaters.length; __updatersIndex++) {
        _nodes[__index].updaters[__updatersIndex].update(st);
      }
    }
    _nodes.length = items.length;
    for (let i = prevCount; i < items.length; i++) {
      _createChildren(i);
    }
  }
});"
`;
